feat: added C implementation for `@stdlib/stats/base/dists/pareto-type1/quantile`

DhruvArvindSingh opened the pull request:
Resolves #3864.

## Description

> What is the purpose of this pull request?

This pull request:

-   #3864 

## Related Issues

> #3864 

This pull request:

-   resolves #3864  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

DhruvArvindSingh committed:
feat: added C implementation for @stdlib/stats/base/dists/pareto-type1/quantile

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter committed:
Merge branch 'develop' into develop_#3864

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/variance/include/stdlib/stats/base/dists/pareto-type1/variance.h:
```diff
@@ -27,7 +27,7 @@ extern "C" {
 
 /**
-* Returns the variance of a Pareto (Type I) distribution.
+* Evaluates the variance of a Pareto (Type I) distribution.
```
```suggestion
* Returns the variance of a Pareto (Type I) distribution.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/variance/src/main.c:
```diff
@@ -22,10 +22,10 @@
 
 /**
-* Returns the variance of a Pareto (Type I) distribution.
+* Evaluates the variance of a Pareto (Type I) distribution.
```
```suggestion
* Returns the variance of a Pareto (Type I) distribution.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/quantile/benchmark/benchmark.js:
```diff
@@ -45,7 +45,7 @@ bench( pkg, function benchmark( b ) {
 	for ( i = 0; i < len; i++ ) {
 		p[ i ] = uniform( 0.0, 1.0 );
-		alpha[ i ] = uniform( EPS, 100.0 );
+		alpha[ i ] = uniform( 3.0 + EPS, 100.0 );
```
```suggestion
		alpha[ i ] = uniform( EPS, 100.0 );
```

Planeshifter committed:
chore: minor clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/quantile/src/main.c:
```diff
@@ -0,0 +1,49 @@
+
+#include "stdlib/stats/base/dists/pareto-type1/quantile.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/pow.h"
+
+
+/**
```
```suggestion
#include "stdlib/math/base/special/pow.h"

/**
```

Planeshifter committed:
docs: fix up descriptions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
style: remove extra empty line

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/quantile/README.md:
````diff
@@ -157,6 +157,103 @@ for ( i = 0; i < 10; i++ ) {
+### Examples
+
+```c
+#include "stdlib/stats/base/dists/pareto-type1/quantile.h"
+#include <stdlib.h>
+#include <stdio.h>
+static double random_uniform( const double min, const double max ) {
+    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
+    return min + ( v*(max-min) );
+}
+int main( void ) {
````
```suggestion
#include "stdlib/stats/base/dists/pareto-type1/quantile.h"
#include <stdlib.h>
#include <stdio.h>

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
```

Planeshifter committed:
style: add empty lines

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (dismissed):
LGTM after clean-up.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/quantile/README.md:
```diff
@@ -157,6 +157,105 @@ for ( i = 0; i < 10; i++ ) {
+
+</section>
+
+<!-- /.c -->
+
+
 <!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
```
```suggestion
<!-- /.c -->

<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
```

Planeshifter committed:
style: remove extra empty line

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/quantile/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+#include "stdlib/stats/base/dists/pareto-type1/quantile.h"
+#include "stdlib/math/base/napi/ternary.h"
+
+// cppcheck-suppress shadowFunction
```
```suggestion
```

Planeshifter committed:
chore: remove unused directive

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
